### PR TITLE
drivers/dac: clarified doc and named return values

### PR DESCRIPTION
**Pull request by haukepetersen**
- added more comprehensive doc to dac_init and dac_set
- named return values for dac_init
- use named return values in existing implementations

### CODE REVIEW

File: drivers/include/periph/dac.h [hunk 1]
```diff
@@ -78,15 +86,23 @@ typedef unsigned int dac_t;
...
  */
 int8_t dac_init(dac_t line);
 
 /**
- * @brief Write a value onto DAC Device on a given Channel.
+ * @brief   Write a value onto DAC Device on a given Channel
+ *
+ * The value is always given as 16-bit value and is internally scaled to the
+ * actual resolution that the DAC unit provides (e.g. 12-bit). So to get the
+ * maximum output voltage, this function hast to be called with @p value set to
```
Comment (aabadie) [inline drivers/include/periph/dac.h]: s/hast/has/

**Comment by haukepetersen** [comment]
fixed typo and squashed it right in.

**Comment by haukepetersen** [comment]
@aabadie do you give your approval?

**Comment by aabadie** [comment]
and go

**Pull request merged**